pkg/api/k8s_versions.go

```diff
@@ -64,206 +49,244 @@ const (
+			common.ClusterAutoscalerAddonName: "cluster-autoscaler:v1.3.9",
+		},
+		"1.10": {
+			common.AddonResizerComponentName:  "addon-resizer:1.8.4",
+			common.MetricsServerAddonName:     "metrics-server-amd64:v0.2.1",
+			common.AddonManagerComponentName:  "kube-addon-manager-amd64:v8.9.1",
+			common.ReschedulerAddonName:       "rescheduler:v0.3.1",
+			common.ClusterAutoscalerAddonName: "cluster-autoscaler:v1.2.5",
+			common.DNSSidecarComponentName:    "k8s-dns-sidecar-amd64:1.14.8",
+		},
+		"1.9": {
+			common.AddonResizerComponentName:  "addon-resizer:1.8.4",
+			common.MetricsServerAddonName:     "metrics-server-amd64:v0.2.1",
+			common.AddonManagerComponentName:  "kube-addon-manager-amd64:v8.6",
+			common.ReschedulerAddonName:       "rescheduler:v0.3.1",
+			common.ClusterAutoscalerAddonName: "cluster-autoscaler:v1.1.2",
+			common.DNSSidecarComponentName:    "k8s-dns-sidecar-amd64:1.14.7",
+		},
+		"1.8": {
+			common.AddonResizerComponentName: "addon-resizer:1.7",
+			common.HeapsterAddonName:         "heapster-amd64:v1.5.1",
+			common.MetricsServerAddonName:    "metrics-server-amd64:v0.2.1",
+			common.KubeDNSAddonName:          "k8s-dns-kube-dns-amd64:1.14.13",
+			common.AddonManagerComponentName: "kube-addon-manager-amd64:v8.6",
+			common.DNSMasqComponentName:      "k8s-dns-dnsmasq-nanny-amd64:1.14.8",
+			common.ReschedulerAddonName:      "rescheduler:v0.3.1",
+		},
+		"1.7": {
+			"dashboard":                      "kubernetes-dashboard-amd64:v1.6.3",
+			common.AddonResizerComponentName: "addon-resizer:1.7",
+			common.HeapsterAddonName:         "heapster-amd64:v1.5.1",
+			common.MetricsServerAddonName:    "metrics-server-amd64:v0.2.1",
+			common.KubeDNSAddonName:          "k8s-dns-kube-dns-amd64:1.14.5",
+			common.AddonManagerComponentName: "kube-addon-manager-amd64:v8.6",
+			common.DNSMasqComponentName:      "k8s-dns-dnsmasq-nanny-amd64:1.14.5",
+			common.ReschedulerAddonName:      "rescheduler:v0.3.1",
+		},
+		"1.6": {
+			"dashboard":                      "kubernetes-dashboard-amd64:v1.6.3",
+			common.AddonResizerComponentName: "addon-resizer:1.7",
+			common.HeapsterAddonName:         "heapster-amd64:v1.3.0",
+			common.MetricsServerAddonName:    "metrics-server-amd64:v0.2.1",
+			common.KubeDNSAddonName:          "k8s-dns-kube-dns-amd64:1.14.5",
+			common.AddonManagerComponentName: "kube-addon-manager-amd64:v6.5",
+			common.DNSMasqComponentName:      "k8s-dns-dnsmasq-nanny-amd64:1.14.5",
+			common.ReschedulerAddonName:      "rescheduler:v0.3.1",
+		},
+	},
 }
 
-func getKubeConfigs() map[string]map[string]string {
+type getK8sVersionComponentsOverrides func(string) map[string]string
+
+func GetK8sComponentsByVersionMap(k *KubernetesConfig) map[string]map[string]string {
+	var overrides getK8sVersionComponentsOverrides
+	switch k.KubernetesImageBaseType {
+	case common.KubernetesImageBaseTypeGCR:
+		overrides = getVersionOverridesGCR
+	case common.KubernetesImageBaseTypeMCR:
```

Comment:
Providing an MCR example here for how we'll evolve this, even though the codebase as-is doesn't yet support the "mcr" `KubernetesImageBaseType` val